paymentsheet/src/test/java/com/stripe/android/paymentelement/confirmation/epms/ExternalPaymentMethodConfirmationDefinitionTest.kt

```diff
@@ -0,0 +1,257 @@
+        val result = definition.toPaymentConfirmationResult(
+            confirmationOption = EPM_CONFIRMATION_OPTION,
+            intent = PAYMENT_INTENT,
+            deferredIntentConfirmationType = null,
+            result = PaymentResult.Failed(exception),
+        )
+
+        assertThat(result).isInstanceOf<ConfirmationHandler.Result.Failed>()
+
+        val failedResult = result.asFailed()
+
+        assertThat(failedResult.cause).isEqualTo(exception)
+        assertThat(failedResult.message).isEqualTo(R.string.stripe_something_went_wrong.resolvableString)
+        assertThat(failedResult.type).isEqualTo(ConfirmationHandler.Result.Failed.ErrorType.ExternalPaymentMethod)
+    }
+
+    @Test
+    fun `'toPaymentConfirmationResult' should return 'Canceled' when 'PaymentResult' is 'Canceled'`() = runTest {
+        val definition = createExternalPaymentMethodConfirmationDefinition()
+
+        val result = definition.toPaymentConfirmationResult(
+            confirmationOption = EPM_CONFIRMATION_OPTION,
+            intent = PAYMENT_INTENT,
+            deferredIntentConfirmationType = null,
+            result = PaymentResult.Canceled,
+        )
+
+        assertThat(result).isInstanceOf<ConfirmationHandler.Result.Canceled>()
+
+        val canceledResult = result.asCanceled()
+
+        assertThat(canceledResult.action).isEqualTo(ConfirmationHandler.Result.Canceled.Action.None)
+    }
+
+    @Test
+    fun `'Fail' action should be returned if EPM handler is not set and report error`() = runTest {
+        val errorReporter = FakeErrorReporter()
+        val definition = createExternalPaymentMethodConfirmationDefinition(errorReporter)
+
+        val action = definition.action(
+            confirmationOption = EPM_CONFIRMATION_OPTION,
+            intent = PAYMENT_INTENT,
+        )
+
+        assertThat(action).isInstanceOf<ConfirmationDefinition.ConfirmationAction.Fail<Unit>>()
+
+        val failAction = action.asFail()
+
+        assertThat(failAction.cause).isInstanceOf<IllegalStateException>()
+        assertThat(failAction.cause.message).isEqualTo(
+            "externalPaymentMethodConfirmHandler is null." +
+                " Cannot process payment for payment selection: paypal"
+        )
+        assertThat(failAction.message).isEqualTo(R.string.stripe_something_went_wrong.resolvableString)
+        assertThat(failAction.errorType).isEqualTo(
+            ConfirmationHandler.Result.Failed.ErrorType.ExternalPaymentMethod,
+        )
+
+        assertThat(errorReporter.getLoggedErrors()).containsExactly(
+            ErrorReporter.ExpectedErrorEvent.EXTERNAL_PAYMENT_METHOD_CONFIRM_HANDLER_NULL.eventName
+        )
+    }
+
+    @Test
+    fun `'Launch' action should be returned if EPM handler is set & report launch`() = runTest {
+        ExternalPaymentMethodInterceptor.externalPaymentMethodConfirmHandler =
+            ExternalPaymentMethodConfirmHandler { _, _ ->
+                // Do nothing
```

Comment:
Is this expected to be called? If so, we should assert that it was, with the right params. If it's not, we should throw here.